interfaces/Matlab_Toolbox_Revamp/Class/FlowDevice.m

```diff
@@ -0,0 +1,154 @@
+                                'Valve'};
+                typ = device_types(typ);
+            end
+            
+            x.type = typ;
+            x.id = calllib(ct, 'flowdev_new2', typ);
+%             if x.id < 0
+%                 error(geterr);
+%             end
+            x.upstream = -1;
+            x.downstream = -1;
+        end
+        
+        %% Utility methods
+
+        function clear(f)
+            % Clear the specified flow device from memory.
+            checklib;
+            calllib(ct, 'flowdev_del', f.id);
+        end
+        
+        %% FlowDevice methods
+
+        function install(f, upstream, downstream)
+            % Install a flow device between reactors or reservoirs.
+            %
+            % parameter upstream:
+            %    Upsteram 'Reactor' or 'Reservoir'.
+            % parameter downstream:
+            %    Downstream 'Reactor' or 'Reservoir'.
+            
+            checklib;
+            if nargin == 3
+                if ~isa(upstream, 'Reactor') || ~isa(downstream, 'Reactor')
+                    error(['Flow devices can only be installed between',...
+                           'reactors or reservoirs']);
+                end
+                i = upstream.id;
+                j = downstream.id;
+                calllib(ct, 'flowdev_install', f.id, i, j);
+%                 if ok < 0
+%                     error(geterr)
+%                 end
+            else error('install requires 3 arguments');
+            end
+        end
+        
+        function mdot = massFlowRate(f, time)
+            % Get the mass flow rate at a given time.
+            %
+            % parameter time:
+            %    Time at which the mass flow rate is desired.
+            % return:
+            %    The mass flow rate through the flow device at the given
+            %    time.
+            
+            checklib;
+            if nargin == 1
+                mdot = calllib(ct, 'flowdev_massFlowRate2', f.id);
+            else
+                warning(['"Time" argument to massFlowRate is deprecated', ...
+                         'and will be removed after Cantera 2.5.']);
+                mdot = calllib(ct, 'flowdev_massFlowRate', f.id, time);
+            end
+        end
+        
+        function setFunction(f, mf)
+            % Set the time function with class 'func'.
+            % 
+            % parameter mf:
+            %    Instance of class 'func'.
+            
+            checklib;
+            if strcmp(f.type, 'MassFlowController')
+                k = calllib(ct, 'flowdev_setTimeFunction', f.id, ...
+                            mf.id);
+%                 if k < 0
+%                     error(geterr);
+%                 end
+            else
+                error('Mass flow rate can only be set for mass flow controllers.');
+            end
+        end
+        
+        function setMassFlowRate(f, mdot)
```

Comment:
Does it make sense to create a child class called `MassFlowController` to implement this specific function? I'm not sure how class inheritance works in Matlab.